operator/pkg/reconciliation/reconcile_reaper.go

```diff
@@ -0,0 +1,235 @@
+		{Name: "admin", ContainerPort: ReaperAdminPort, Protocol: "TCP"},
+	}
+
+	container := corev1.Container{
+		Name: ReaperContainerName,
+		Image: getReaperImage(dc),
+		ImagePullPolicy: getReaperPullPolicy(dc),
+		Ports: ports,
+		LivenessProbe: probe(ReaperAdminPort, ReaperHealthCheckPath, int(60 * dc.Spec.Size), 10),
+		ReadinessProbe: probe(ReaperAdminPort, ReaperHealthCheckPath, 30, 15),
+		Env: []corev1.EnvVar{
+			{Name: "REAPER_STORAGE_TYPE", Value: "cassandra"},
+			{Name: "REAPER_ENABLE_DYNAMIC_SEED_LIST", Value: "false"},
+			{Name: "REAPER_DATACENTER_AVAILABILITY", Value: "SIDECAR"},
+			{Name: "REAPER_SERVER_APP_PORT", Value: strconv.Itoa(ReaperUIPort)},
+			{Name: "REAPER_SERVER_ADMIN_PORT", Value: strconv.Itoa(ReaperAdminPort)},
+			{Name: "REAPER_CASS_CLUSTER_NAME", Value: dc.ClusterName},
+			{Name: "REAPER_CASS_CONTACT_POINTS", Value: fmt.Sprintf("[%s]", dc.GetSeedServiceName())},
+			{Name: "REAPER_AUTH_ENABLED", Value: "false"},
+			{Name: "REAPER_JMX_AUTH_USERNAME", Value: ""},
+			{Name: "REAPER_JMX_AUTH_PASSWORD", Value: ""},
+		},
+	}
+
+	return container
+}
+
+func getReaperImage(dc *api.CassandraDatacenter) string {
+	if len(dc.Spec.Reaper.Image) == 0 {
+		return ReaperDefaultImage
+	}
+	return dc.Spec.Reaper.Image
+}
+
+func getReaperPullPolicy(dc *api.CassandraDatacenter) corev1.PullPolicy {
+	if len(dc.Spec.Reaper.ImagePullPolicy) == 0 {
+		return ReaperDefaultPullPolicy
+	}
+	return dc.Spec.Reaper.ImagePullPolicy
+}
+
+func (rc *ReconciliationContext) CheckReaperSchemaInitialized() result.ReconcileResult {
+	// Using a job eventually get replaced with calls to the mgmt api once it has support for
+	// creating keyspaces and tables.
+
+	rc.ReqLogger.Info("reconcile_reaper::CheckReaperSchemaInitialized")
+
+	if !rc.Datacenter.Spec.Reaper.Enabled {
+		return result.Continue()
+	}
+
+	jobName := getReaperSchemaInitJobName(rc.Datacenter)
+	schemaJob := &v1batch.Job{}
+
+	err := rc.Client.Get(rc.Ctx, types.NamespacedName{Namespace: rc.Datacenter.Namespace, Name: jobName}, schemaJob)
+	if err != nil && errors.IsNotFound(err) {
+		// Create the job
+		schemaJob := buildInitReaperSchemaJob(rc.Datacenter)
+		rc.ReqLogger.Info("creating Reaper schema init job", ReaperSchemaInitJob, schemaJob.Name)
+		if err := setControllerReference(rc.Datacenter, schemaJob, rc.Scheme); err != nil {
+			rc.ReqLogger.Error(err, "failed to set owner reference", ReaperSchemaInitJob, schemaJob.Name)
+			return result.Error(err)
+		}
+		if err := rc.Client.Create(rc.Ctx, schemaJob); err != nil {
+			rc.ReqLogger.Error(err, "failed to create job", ReaperSchemaInitJob, schemaJob.Name)
+			return result.Error(err)
+		} else {
+			return result.RequeueSoon(2)
+		}
+	} else if err != nil {
+		return result.Error(err)
+	} else if jobFinished(schemaJob) {
+		return result.Continue()
+	} else {
+		return result.RequeueSoon(2)
+	}
+}
+
+func buildInitReaperSchemaJob(dc *api.CassandraDatacenter) *v1batch.Job {
```

Comment:
move this over to the constructor.go file?

Reply:
I moved the `buildInitReaperSchemaJob` function. Did you want the `buildReaperContainer` function moved as well?